x-pack/plugin/sql/qa/jdbc/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/JdbcTestUtils.java

```diff
@@ -104,7 +113,22 @@ static int extractNanosOnly(long nanos) {
         return (int) (nanos % 1_000_000_000);
     }
 
+    static <T> Set<T> randomSet(int minSetSize, int maxSetSize, Supplier<T> valueConstructor) {
+        final int size = randomIntBetween(minSetSize, maxSetSize);
+        Set<T> set = new HashSet<>();
```

Comment:
Why not putting a `size` to the HashSet constructor, since it's already known. To help with this a bit, we have in CollectionUtils the method `mapSize`.

Reply:
`CollectionUtils` is in `ql`, which isn't a dependency. Not sure if optimising a random-size-driven test would be worth, but still, I could add the method ("formula") here if you think it'd be better that way.

Reply:
Not worth adding ql as a dependency for this.